### DISCUSSION TITLE
Add possibility to deposit signature for signed RDP connections

### DISCUSSION THREAD

**Discussion by andy1463** [1Remote/1Remote#567] [Ideas]
If you open a RDP connection in MSTSC mode you get the message "the publisher of this remote connection can't be identified"
![1remote-4](https://github.com/1Remote/1Remote/assets/128373669/9a7f241d-7260-45b8-aa5d-f178b4cedb55)

This message will not appear if you sign the RDP file as trusted publisher.
An option for adding either the signature of a signed RDP file to the config or possibility to import signed RDP files would be great.

**Comment by VShawn** [comment]
i have no idea how to get a signed rdp file and I am not sure if 1Remote can signed a rdp file since, do you have any suggestion?

**Reply by andy1463** [reply]
You can sign a rdp file with rdpsign.exe 

usage:
rdpsign.exe /sha256 THUMBPRINT_OF_CERTIFICATE RDPFILE.rdp

this adds three lines to rdp file:
alternate full address, signscope and signature
i.e.

alternate full address:s:192.168.0.4:3389

signscope:s:Full Address,Alternate Full Address,Use Redirection Server Name,Negotiate Security Layer,AutoReconnection Enabled,GatewayUsageMethod,GatewayProfileUsageMethod,GatewayCredentialsSource,PromptCredentialOnce,RemoteApplicationMode,Prompt For Credentials,Authentication Level,AudioMode,RedirectDrives,RedirectPrinters,RedirectCOMPorts,RedirectSmartCards,RedirectPOSDevices,RedirectClipboard,DevicesToRedirect,RedirectDirectX,RDGIsKDCProxy

signature:s:ArABAAEAAAAzCrAAMIILswYJKoZIhsNAQcCoIILdfCCCxwCAQExDzANBglghkgB  ZQMEAgEFADALBgkqtr9w0BBwGggglvMIIDdzCCAl+gAwIBAgIQdfbXM9uUD6lO  hr3pDL3yRzANBgkfkiG9w0BAsdDBCMRUwEwYKCZImiZPyLGQBGRYFbG9jYWwx  EzARBgoJkiaJk/IsZAEZFgN6bfaDASBgNVBAMTC3ptdi1EQy1DQS0xMB4XDTIz  NACUUuqkieryETnlL16NTokfpXf9u6nalymNswXqXMn/W82zDvR06xFqporihRe  PgChkUrn2oJ21y6HsytUSBdGyq9D9WDBeWleYBwtw7vF7DTSWK+A8WYd6XqxT3  AwEAAaNpMGcwEwYJKwYBBAGCNxQCBAYeBABDAEEwDgYDVR0PAQH/BAQDAgGGMA8G   CSsGAQQBgjcVAQQDAgEAMA0GCSqGSIb3DQEBCwUAA4IBAQAEltEpb/Sdi2ww/xDg  sAFJhubl35j8y4g6iWeEjBfgJBOjBNHx6itEZuk4MtyLYahV7RtRSfB2sgB  0i4smn3cIHERoIZ7orZrztyG0Mk3azJyThNN+9f9jZ0LxNbXiBmvROzAFWgTJ66j  js3KMIIF8DCCBNigjwIcAAAAAJcpTj11NVovAAAAAAAAjANBgkqhkiG9w0B  AQsFADBCMRUwEwYKCZImiZPyLGQBGRYFbG9jYWwxEzARBgoJkiaJk/IsZAEZFgN6  bXYxFDASBgNVBAMTC3ptzui1EQy1DQS0xMB4XDTIzMTExNzA4NDIwM1oXDTI0MTEx  NjA4NDIwM1owVDEVMBMGCfmT8ixkARkWBWxvY2FsMRMwEQYKCZImiZPyLGQB  GRYDem12MQ4wDAYDVQQDfawVVc2VyczEWMBQGA1UEAxMNQWRtaW5pc3RyYXRvcjCC  ASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBANySbHPhwU7gBEl9T/ehlUPp  1pEmIkicpAnDj72tP/0B/I24rV3ZDULMlYCt4WtAnFWv2J5Bh3QxkDm1WKb/0PEC  AwEAAaOCAsswggLHMD0GCSsGAQQBghBwQwMC4GJisGAQQBgjcVCITN7XSHs7Bc  gamTDoSJ0yqFnop0gUCQlnyCmO5dAgFkAgECMBMGA1UdJQQMMAoGCCsGAQUFBwMD  MA4GA1UdDwEB/wQEAwIHgDAbBgkrBgEEAYI3FQoEDjAMMAoGCCsGAQUFBwMDMB0G  A1UdDgQWBBSN/Nh2kfKXG4LUfu0e7LFuqDqy3MIG0oIGxoIGuhoGrbGRhcDovLy9D  Tj16bXYtREMtQ0EtMSxDTj1kYyxDTj1DRFAsQ049UHVibGljJTIwS2V5JTIwU2Vy  dmljZXMsQ049U2VydmljZXMsQ049Q29rmlndXJhdGlvbixEQz16bXYsREM9bG9j  ZXklMjBTZXJ2aWNlsDTj1TZXJ2aWNlcyxDTsDb25maWd1cmF0aW9uLERDPXpt  dixEQz1sb2NhbD9jQUNlcnRpZmljYXRlP2Jhc2U/b2JqZWN0Q2xhc3M9Y2VydGlm  aWNhdGlvbkF1dfcml0eTAyBgNVHREEKzApoCcGCisGAQQBgjcUAgOgGQwXYWRt  aW5pc3RyYXRvckB6bXYubG9jYfwwTQYJKwYBBAGCNxkCBEAwPqA8BgorBgEEAYI3  GQIBoC4ELFMtMS01LTgTI0NDUxOTYzOTktMTY2MTIwOTMzOS03MDI2Mjg3Mzkt  NTAwMA0GCSqGSIb3DQEBCwUAA4IBhgDTzrPpVbLnwbsM4eMD3+PEtG14M8Orhy0  kMKqbYVvjL2MjBrpyaZ4TQHTAzQH1t2E0JvqdNadsOrGXt0lclWA/KuTlrmsEr  WxLeK2j77foQBw2af1XFLj2guWHI7fqbf6H+/5CQYYXXWKdWTSugDHA9g7T0  5A3T64bRx/wZbZsSls3bq8D/pUKAqYrPXSPNi5R2r5zEjNByWQcKuaPSX4g35m  aXXpd562mn7EhdiTNYMD+j2ZEexs+gm0sbSz8YrFARuRj9Rg5Fm68UVomiEMIy0b  JM2D76wct09Xc54guX9+fi4oSzlNdODdG+A4x3ex8rOi9FHWgbMYIBhDCCAYAC  AQEwWTBCMRUwEwYKCZImiZPyLGQBGRYFbG9jYWwxEzARBgoJkiaJk/IsZAEZFgN6   OUH1WeCbatTQZj/JOz/b91J6+YyRnwrWK6oDy5n2zbxxyCfoxCCQaPWsZzwDuw=

There is already an option "preview rdp file". Maybe you can add an option to manually add the lines for signed rdp files?

**Reply by VShawn** [reply]
sounds easy to achieve. 

but according to https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/rdpsign

We need a `hash` to finsh the sign, where we can get it? And if it requires the user to manually install a certificate and then specify the hash of the certificate in 1Remote, the user experience would be poor. I would prefer to have the user check the "do not ask me again" option.

**Reply by andy1463** [reply]
Enterprise environments often have PKI installed and certificates will be rolled out via group policies. So it won't be necessary to install a certificate manually.
Maybe it is an idea to add an additional field for hash value and sign the connection if mstsc mode is checked?
If you have a lot of rdp connections you can sign them with the same certificate, so you only need the hash value one time. Should be easy for admins to read the hash value from the certificate.

![1rem-4](https://github.com/1Remote/1Remote/assets/128373669/1ca4387b-c5d3-424e-b716-e9f7ee80959d)

**Reply by VShawn** [reply]
simply adding the option is easy, but the concern is how many people will be able to extract the hash on their own. it would be helpful to provide a hint on how to obtain the hash, and there should also be instructions for certificate installation for those non-enterprise users. However with all this above, things becomes more complex and requires more time to complete.

**Reply by VShawn** [reply]
And I'm also doubting how many users would benefit from this option. Personally, I have never been aware of this signature option before, and it has never caused any inconvenience. Lastly, for the sake of user experience, I would prefer not to add too many complex options in the configuration menu that the average person may not understand. There are already plenty of settings options available.

**Reply by andy1463** [reply]
I understand your argumentation. 
Maybe an idea for future versions. Add something like "expert mode" which is shielded by default and if you activate it you can add more specific options.
